electricpy/visu.py

```diff
@@ -321,4 +325,209 @@ def get_output_power(self):
         # low slip
         p_x_2 = center_x - self.radius * _np.cos(beta_1)
         p_y_2 = center_y + self.radius * _np.sin(beta_1)
-        return [p_x_1, p_y_1], [p_x_2, p_y_2]
\ No newline at end of file
+        return [p_x_1, p_y_1], [p_x_2, p_y_2]
+
+class PowerCircle:
+    r"""
+    Plot Power Circle Diagram of Transmission System.
+
+    This class is designed to plot the power circle diagram of a transmission
+    system both sending and reciving ends.
+
+    Parameters
+    ----------
+    type: str "sending" or "receiving"
+
+    Vr: complex Receiving End Voltage
+
+    Vs: complex Sending End Voltage
+
+    power_factor: float Power Factor
+
+    Pr: float Receiving End Real Power
+
+    Qr: float Receiving End Reactive Power
+
+    Sr: float Receiving End Total Power
+
+    Ps: float Sending End Real Power
+
+    Qs: float Sending End Reactive Power
+
+    Ss: float Sending End Total Power
+
+    power_factor: float Power Factor
+
+    A: float ABCD Matrix
+    
+    B: float ABCD Matrix
+    
+    C: float ABCD Matrix
+    
+    D: float ABCD Matrix
+
+    """
+
+    def __init__(self, type, Vr: complex = None, Vs: complex = None, power_factor: float = None,
```

Comment:
I see that almost all of these variables have a default. If some of them are optional, they should be declared as such in the docstring.

e.g.,

```python
Vr:    complex, optional
       Some description about the quantity. Default: None
```

Reply:
Sir i cant keep them optional because if user want 

receiving end power circle he should provide 
  `Vr`, `A`, `B` 

and for sending end power circle
`Vs` , `B` , `D`

can you help me around with it